### PR TITLE
Hw

### PR DESCRIPTION
**Pull request by Richaa2**

### CODE REVIEW

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert drivers by carId " + car.getId(), e);
+        }
+    }
+
+    private void deleteRelationsCars_Drivers(Car car) {
```
Comment (aaananas) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:176]:
```suggestion
    private void deleteRelationsCarsDrivers(Car car) {
```

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturerFromResultSet(resultSet));
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list"
+                    + " of cars from manufacturer_DB.", e);
+        }
+        for (Car car : cars) {
+            if (car != null) {
```
Comment (aaananas) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:82]:
```suggestion
```

File: src/main/java/mate/jdbc/service/CarServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    private CarDao carDao = new CarDaoImpl();
+
+    @Override
+    public Car create(Car car) {
+        return carDao.create(car);
+    }
+
+    @Override
+    public Car get(Long id) {
+        return carDao.get(id).orElseThrow(RuntimeException::new);
```
Comment (aaananas) [inline src/main/java/mate/jdbc/service/CarServiceImpl.java:24]: message should be informative

File: src/main/java/mate/jdbc/service/CarServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        return carDao.getAll().stream().filter((c) -> {
+            for (Driver driver : c.getDrivers()) {
+                if (driver.getId().equals(driverId)) {
+                    return true;
+                }
+            }
+            return false;
+        }).collect(Collectors.toList());
```
Comment (aaananas) [inline src/main/java/mate/jdbc/service/CarServiceImpl.java:68]: you should get cars by driver from db - with sql query

File: src/main/java/mate/jdbc/service/CarServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        List<Driver> drivers = car.getDrivers();
+        drivers.add(driver);
+        car.setDrivers(drivers);
+        carDao.update(car);
```
Comment (aaananas) [inline src/main/java/mate/jdbc/service/CarServiceImpl.java:47]:
```suggestion
        car.getDrivers().add(driver);
        carDao.update(car);
```

File: src/main/java/mate/jdbc/service/CarServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        List<Driver> drivers = car.getDrivers();
+        drivers.remove(driver);
+        car.setDrivers(drivers);
+        carDao.update(car);
```
Comment (aaananas) [inline src/main/java/mate/jdbc/service/CarServiceImpl.java:55]:
```suggestion
        car.getDrivers().remove(driver);
        carDao.update(car);
```

**Review by aaananas — CHANGES_REQUESTED** [review]

**Review by aleksandr-bevza — APPROVED** [review]

**Pull request closed without merging**